twindb_infrastructure/providers/aws.py

```diff
@@ -140,35 +165,16 @@ def launch_ec2_instance(instance_profile, region=AWS_REGIONS[0],
     except KeyError:
         pass
 
-    cmd.append(json.dumps(device_mappings))
-
-    log.debug("Executing: %s" % ' '.join(cmd))
-    aws_env = {
-        'AWS_ACCESS_KEY_ID': aws_access_key_id,
-        'AWS_SECRET_ACCESS_KEY': aws_secret_access_key,
-        'AWS_DEFAULT_REGION': region,
-        'PATH': os.environ['PATH'],
-        'LC_ALL': 'en_US.UTF-8',
-        'LC_CTYPE': 'UTF-8',
-        'LANG': 'en_US'
-    }
-
-    cout, cerr = None, None
+    client_args['BlockDeviceMappings'] = device_mappings
+    response = None
     try:
-        aws_process = Popen(cmd, stdout=PIPE, stderr=PIPE, env=aws_env)
-        cout, cerr = aws_process.communicate()
-
-        if aws_process.returncode != 0:
-            log.error('Failed to execute %s' % ' '.join(cmd))
-            log.error(cerr)
-            return None
-    except OSError as err:
+        response = client.run_instances(**client_args)
+    except ClientError as err:
         log.error('Failed to execute %r' % cmd)
```

Comment:
I would still raise `AwsError` here and let the outer function to exit if it deems necessary